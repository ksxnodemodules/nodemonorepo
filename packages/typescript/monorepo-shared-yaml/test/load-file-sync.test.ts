import createVirtualEnvironment from './.lib/virtual-env'
import subject from '../index'

const {apply} = createVirtualEnvironment('load.yaml')

it('matches snapshot', apply(async () => {
  expect(subject.loadFileSync('root/source.yaml')).toMatchSnapshot()
}))

it('throws error matching snapshot', apply(async () => {
  expect(() => subject.loadFileSync('root/not-yaml')).toThrowErrorMatchingSnapshot()
}))
