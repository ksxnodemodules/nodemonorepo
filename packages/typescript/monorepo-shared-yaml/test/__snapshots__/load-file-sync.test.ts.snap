// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "hello": "world",
}
`;

exports[`throws error matching snapshot 1`] = `
"Failed to parse 'root/not-yaml': incomplete explicit mapping pair; a key node is missed; or followed by a non-tabulated empty line at line 1, column 9:
    this: is: not: valid: yaml: syntax:
            ^"
`;
